### PR TITLE
Mount task_local and alloc dir to docker containers

### PR DESCRIPTION
**Pull request by dadgar**
Mounts the tasks local directory and alloc directory to the docker container. Also add better parsing of command arguments with nomad environment variable injection.

### CODE REVIEW

File: client/driver/docker.go [hunk 1]
```diff
@@ -97,12 +100,33 @@ func (d *DockerDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool
 	return true, nil
 }
 
-// We have to call this when we create the container AND when we start it so
-// we'll make a function.
-func createHostConfig(task *structs.Task) *docker.HostConfig {
-	// hostConfig holds options for the docker container that are unique to this
-	// machine, such as resource limits and port mappings
-	return &docker.HostConfig{
+func containerBinds(alloc *allocdir.AllocDir, task *structs.Task) ([]string, error) {
```
Comment (armon) [inline client/driver/docker.go]: Is there a reason this isn't a method of the Docker Driver? It seems unlikely to be shared

File: client/driver/docker.go [hunk 2]
```diff
@@ -97,12 +100,33 @@ func (d *DockerDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool
...
+	}
+
+	return []string{
+		fmt.Sprintf("%s:%s", shared, allocdir.SharedAllocName),
+		fmt.Sprintf("%s:%s", local, allocdir.TaskLocal),
+	}, nil
+}
+
+// createContainer initializes a struct needed to call docker.client.CreateContainer()
+func createContainer(ctx *ExecContext, task *structs.Task, logger *log.Logger) (docker.CreateContainerOptions, error) {
```
Comment (armon) [inline client/driver/docker.go]: Similarly, would this ever be shared?

File: scripts/build.sh [hunk 3]
```diff
@@ -44,6 +44,7 @@ gox \
     -arch="${XC_ARCH}" \
     -osarch="!linux/arm !darwin/386" \
     -ldflags "-X main.GitCommit ${GIT_COMMIT}${GIT_DIRTY}" \
+    -cgo \
```
Comment (armon) [inline scripts/build.sh]: I thought we specifically were avoiding cgo?

Comment (dadgar) [inline scripts/build.sh]: It can't be. user.Lookup which is part of the stdlib uses cgo.

File: client/driver/docker.go (hunk 2 above)
Comment (dadgar) [inline client/driver/docker.go]: Yeah I can update these. Just an artifact of how it was written.

**Pull request merged**